[BUG] Can't publish to NPM via Github workflow

I have a freshly [set up project](https://github.com/superluminar-io/config-kitty/blob/main/.projenrc.ts). It fails when publishing to NPM.

```
Run npx -p publib@latest publib-npm
npm warn exec The following package was not found and will be installed: publib@0.2.886
Publishing under the following dist-tag: latest
npm warn tarball tarball data for file:dist/js/**.tgz (null) seems to be corrupted. Trying again.
npm warn tarball tarball data for file:dist/js/**.tgz (null) seems to be corrupted. Trying again.
npm error code ENOENT
npm error syscall open
npm error path /home/runner/work/config-kitty/config-kitty/dist/js/**.tgz
npm error errno -2
npm error enoent ENOENT: no such file or directory, open '/home/runner/work/config-kitty/config-kitty/dist/js/**.tgz'
npm error enoent This is related to npm not being able to find a file.
npm error enoent

npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2024-09-17T15_35_25_274Z-debug-0.log
ERROR
Error: Process completed with exit code 1.
```

https://github.com/superluminar-io/config-kitty/actions/runs/10906325955/job/30267363705


What exactly is going wrong here? It seems that the publishing component is looking for a `*.tgz`, but there is none.

--- a/src/javascript/node-project.ts
+++ b/src/javascript/node-project.ts
@@ -834,7 +834,13 @@ export class NodeProject extends GitHubProject {
     // add a bundler component - this enables things like Lambda bundling and in the future web bundling.
     this.bundler = new Bundler(this, options.bundlerOptions);
 
-    if (options.package ?? true) {
+    const shouldPackage = options.package ?? true;
+    if (release && !shouldPackage) {
+      this.logger.warn(
+        "When `release` is enabled, `package` must also be enabled as it is required by release. Force enabling `package`."
+      );
+    }
+    if (release || shouldPackage) {
       this.packageTask.exec(`mkdir -p ${this.artifactsJavascriptDirectory}`);
 
       const pkgMgr =

--- a/src/typescript/typescript.ts
+++ b/src/typescript/typescript.ts
@@ -794,11 +794,14 @@ class SampleCode extends Component {
  */
 export class TypeScriptAppProject extends TypeScriptProject {
   constructor(options: TypeScriptProjectOptions) {
+    // Releasing and packaging are coupled. If one is disabled, disable the other by default.
+    const shouldRelease = options.release ?? options.releaseWorkflow ?? false;
+
     super({
+      release: shouldRelease,
+      package: shouldRelease,
       allowLibraryDependencies: false,
-      releaseWorkflow: false,
       entrypoint: "", // "main" is not needed in typescript apps
-      package: false,
       ...options,
     });
   }

--- a/test/cdk8s/cdk8s-app-project-ts.test.ts
+++ b/test/cdk8s/cdk8s-app-project-ts.test.ts
@@ -178,7 +178,7 @@ test("upgrade task ignores pinned versions", () => {
     cdk8sVersionPinning: true,
     cdk8sCliVersionPinning: true,
     cdk8sPlusVersionPinning: true,
-    releaseWorkflow: true,
+    release: true,
     constructsVersion: "3.3.75",
   });
   const tasks = synthSnapshot(project)[TaskRuntime.MANIFEST_FILE].tasks;

--- a/test/typescript/typescript.test.ts
+++ b/test/typescript/typescript.test.ts
@@ -6,6 +6,7 @@ import { Transform } from "../../src/javascript";
 import {
   mergeTsconfigOptions,
   TsJestTsconfig,
+  TypeScriptAppProject,
   TypeScriptProject,
 } from "../../src/typescript";
 import { execProjenCLI, synthSnapshot, withProjectDir } from "../util";
@@ -589,6 +590,35 @@ describe("tsconfigDev", () => {
         })
     ).toThrow(
       "Cannot specify both 'disableTsconfigDev' and 'disableTsconfig' fields."
+    );
+  });
+
+  test("TypeScriptAppProject enables packaging when release is enabled", () => {
+    const project = new TypeScriptAppProject({
+      name: "test",
+      projenrcTs: true,
+      defaultReleaseBranch: "main",
+      release: true,
+    });
+    project.synth();
+
+    expect(project.packageTask.steps.length).not.toBe(0);
+  });
+
+  test("TypeScriptProject force enables packaging when release is enabled", () => {
+    const loggerWarnSpy = jest.spyOn(Logger.prototype, "warn");
+    const project = new TypeScriptAppProject({
+      name: "test",
+      projenrcTs: true,
+      defaultReleaseBranch: "main",
+      release: true,
+      package: false,
+    });
+    project.synth();
+
+    expect(project.packageTask.steps.length).not.toBe(0);
+    expect(loggerWarnSpy).toHaveBeenCalledWith(
+      expect.stringMatching("Force enabling `package`")
     );
   });
 });

